pkg/reconciler/managed/reconciler.go

```diff
@@ -662,6 +664,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(managed),
 	)
 
+	// check the pause annotation and return if it has the value "true"
+	// after logging, publishing an event and updating the SYNC status condition
+	if managed.GetAnnotations()[meta.AnnotationKeyReconciliationPaused] == "true" {
```

Comment:
It might be easier to read if we had a separate `IsPaused(cr) bool` function we call here.

Reply:
Done.